mechafil_jax/locking.py

```diff
@@ -139,20 +155,68 @@ def compute_new_pledge_for_added_power(
     total_qa_power_pib: float,
     baseline_power_pib: float,
     lock_target: float,
+    gamma: float,
+    gamma_weight_type: int,  
 ) -> float:
-    # storage collateral
+    
+    # storage pledge collateral component
     storage_pledge = 20.0 * day_network_reward * (day_added_qa_power_pib / total_qa_power_pib)
-    # consensus collateral
-    normalized_qap_growth = day_added_qa_power_pib / jnp.maximum(total_qa_power_pib, baseline_power_pib)
-    consensus_pledge = jnp.maximum(lock_target * prev_circ_supply * normalized_qap_growth, 0)
+
+    # Compute Simple and Baseline Pledge 
+    simple_consensus_pledge = lock_target * prev_circ_supply * (day_added_qa_power_pib/total_qa_power_pib)
+    simple_consensus_pledge = jnp.maximum(simple_consensus_pledge, 0)
+
+    normalized_qap_growth = day_added_qa_power_pib / jnp.maximum(total_qa_power_pib, baseline_power_pib) # Fraction for Consensus Pledge Component
+    baseline_consensus_pledge = lock_target * prev_circ_supply * normalized_qap_growth
+    baseline_consensus_pledge = jnp.maximum(baseline_consensus_pledge, 0)
+
+    # Compute Simple and Baseline Pledge Contributions to Consensus Pledge for Arithmetic, Geometric, and Harmonic Weightings 
+   
+    #ngamma parameter for weighting 
+    weighting_method = gamma_weight_type #weighting schema (must be arithmetic=0, geometric=1, or harmonic=2)
+
+    consensus_pledge = weighted_consensus(weighting_method, gamma, simple_consensus_pledge, baseline_consensus_pledge)
     # total added pledge
     added_pledge = storage_pledge + consensus_pledge
 
     pledge_cap = day_added_qa_power_pib * (constants.PIB / constants.GIB)  # The # of bytes in a GiB (Gibibyte)
     return jnp.minimum(pledge_cap, added_pledge)
 
+@jax.jit
+def arithmetic_fn(w, simple_consensus_pledge, baseline_consensus_pledge):
+    return (1 - w) * simple_consensus_pledge + w * baseline_consensus_pledge
+
+@jax.jit
+def geometric_fn(w, simple_consensus_pledge, baseline_consensus_pledge):
+    return simple_consensus_pledge ** (1 - w) * baseline_consensus_pledge ** w
+
+@jax.jit
+def harmonic_fn(w, simple_consensus_pledge, baseline_consensus_pledge):
+    return 1 / ((1 - w) * simple_consensus_pledge**-1 + w * baseline_consensus_pledge**-1)
+
+@jax.jit
+def raise_exception(w, simple_consensus_pledge, baseline_consensus_pledge):
```

Comment:
where is this function used? also, I am unclear about jitting functions which raise exceptions. i don't think they should be jitted, but did you find a reference somewhere that does this?